package NIndexerCore;

message TDirectMarkup {
    message TPosition {
        optional uint32 Sent = 1;
        optional uint32 Word = 2;
    }

    message TAttrEntry {
        optional TPosition Position = 1;
        optional bytes Value = 2;
        optional bool NoFollow = 3 [default = false];
    }

    message TAttribute {
        optional uint32 Type = 1;
        optional bytes Name = 2;
        repeated TAttrEntry Entries = 3;
    }

    message TZoneSpan {
        optional TPosition Begin = 1;
        optional TPosition End = 2;
    }

    message TZone {
        optional uint32 Type = 1;
        optional bytes Name = 2;
        repeated TZoneSpan Spans = 3;
    }

    repeated TAttribute Attributes = 1;
    repeated TZone Zones = 2;
}
